lib/azure/armrest/armrest_service.rb

```diff
@@ -217,7 +217,7 @@ def rest_execute(options, http_method = :get, encode = true, max_retries = 3)
               msg = "A rate limit or server side issue has occurred [#{err.http_code}]. Retry number #{tries}."
               Azure::Armrest::Configuration.log.try(:log, Logger::WARN, msg)
               sleep_time = err.response.headers[:retry_after] || 30
-              sleep(sleep_time)
+              sleep(sleep_time.to_i)
```

Comment:
If `:retry_after` is a non integer, this will result in `0` for `sleep_time`.  Is that desired?  Or, should it resort to `30`?

I'm worried that `sleep_time == 0` will result in trying a number of requests over and over without any sleep, and never allowing the API rate limit to reset.

Reply:
@blomquisg as far as I know `:retry_after` will always be a stringified number, which `.to_i` converts to the equivalent integer, e.g. `"25".to_i` is 25, not 0.

Reply:
K.  https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After says that it *could* be a date or a number of seconds.

I'm not sure if Azure has some inconsistency in their value.  If they are sometimes passing back a date, this could be bad for us.  The only way I can see to monitor that is to log out what it's saying for the requested timeout duration.  That might be a good idea anyway in the warn line above this.

Reply:
I've never seen a date passed back for that value, and all of their REST API documentation that I've come across says it will be number of seconds, e.g. https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-request-limits.

Reply:
The thing that's really concerning to me is that this fails today only *sometimes*.  That means that sometimes this current code works. 

That means one of a few things:

The response header value of `retry_after`
a. sometimes comes back as an int, and sometimes comes back as a string, but it's always an integer value (25, or "25")
b. sometimes comes back as empty string, and sometimes comes back as an int ("", or 25)
c. sometimes comes back as a non-int, and sometimes comes back as an int ("Wed, 21 Oct 2015 07:28:00 GMT", or 25)

In any case, I think this should definitely handle the case where the resulting `sleep_time` is `0`.  That way you don't have to care about the special cases for the value in the `retry_after`.